### PR TITLE
allow manual zarr encoding on unchunked dask dimensions

### PR DESCRIPTION
**Pull request by d70-t**
If a dask array is chunked along one dimension but not chunked along
another, any manually specified zarr chunk size should be valid, but
before this patch, this resulted in an error.

 - [x] Tests added
 - [x] Passes `isort . && black . && mypy . && flake8` (only for modified sections)
 - [x] User visible changes (including notable bug fixes) are documented in `whats-new.rst`

### CODE REVIEW

File: doc/whats-new.rst [hunk 1]
```diff
@@ -49,10 +49,11 @@ New Features
 
 Bug fixes
 ~~~~~~~~~
+- Variables which are chunked using dask only along some dimensions can be chunked while storing with zarr along previously
+  unchunked dimensions.
```
Comment (dcherian) [inline doc/whats-new.rst:53]: Please feel free to take credit by adding your name and github id here if you would like to.

Comment (d70-t) [reply doc/whats-new.rst]: Thanks for the hint. I've added my name and the pull request id.

File: doc/whats-new.rst [hunk 2]
```diff
@@ -49,10 +49,11 @@ New Features
 
 Bug fixes
 ~~~~~~~~~
+- Variables which are chunked using dask only along some dimensions can be chunked while storing with zarr along previously
+  unchunked dimensions (:pull:`4312`) `Tobias Kölling <https://github.com/d70-t>`_.
```
Comment (keewis) [inline doc/whats-new.rst:53]: nit: I know the entry below is different (not intentionally, I think), but we usually follow a certain format with these items:
```suggestion
  unchunked dimensions (:pull:`4312`) By `Tobias Kölling <https://github.com/d70-t>`_.
```

**Comment by d70-t** [comment]
Do you know why Read the Docs complains? And if this is related to the PR?

**Comment by keewis** [comment]
could you try merging in `master`?

**Comment by dcherian** [comment]
Thanks @d70-t . I see this is your first PR. Welcome to xarray!

**Pull request merged**